packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -1397,6 +1399,7 @@ class _RenderDecoration extends RenderBox with SlottedContainerRenderObjectMixin
         if (suffixIcon != null) {
           end -= contentPadding.end;
           end += centerLayout(suffixIcon!, end);
+          end += inputToSuffixGap;
```

Comment:
This change is for the `TextDirection.rtl` case, I would also expect a similar change for the prefix (probably at line 1382)

Reply:
@bleroux For both cases we are increasing size of box (line 1008 & 1009). So for `TextDirection.rtl` we are adding gap between `suffix` and `input`. On the other side for `TextDirection.ltr` we are adding gap between `prefix` and `input`. So, I don't think we need to add prefix gap (for `TextDirection.rtl`) or suffix gap (for `TextDirection.ltr`). Because that is already handled by size of the box.

Reply:
The size of the box is correct but the placements in RTL is wrong (too much padding on the left side not enough on the right, see the screenshot I posted).
Testing locally, when adding the logic I pointed, I can get the correct result in RTL but maybe there are other ways to get the correct result.

Reply:
Yup, make sense. Sorry, I didn't checked the images. I have pushed respective changes.